nucypher/blockchain/eth/sol/source/contracts/StakingEscrow.sol

```diff
@@ -327,22 +327,36 @@ contract StakingEscrow is Issuer {
 
     /**
     * @notice Get the value of locked tokens for active stakers in (getCurrentPeriod() + _periods) period
+    * as well as stakers and their locked tokens
     * @param _periods Amount of periods for locked tokens calculation
+    * @return allLockedTokens Sum of locked tokens for active stakers
+    * @return activeStakers Array of stakers and their locked tokens. Stakers addresses stored as uint256
     **/
-    function getAllLockedTokens(uint16 _periods)
-        external view returns (uint256 lockedTokens)
+    function getAllActiveStakers(uint16 _periods)
+        external view returns (uint256 allLockedTokens, uint256[2][] memory activeStakers)
     {
         require(_periods > 0);
+
+        uint256 stakersLength = stakers.length;
```

Comment:
Oh interesting. In solidity, doing it through a local variable is faster?

Reply:
It should slightly decrease size of contract if I remember correctly. Need to check it to be sure

Reply:
If I'm not wrong the explanation is that each access to `stakers.length` is actually an `SLOAD` instruction since `stakers` is and storage variable and its length is located as the first slot, and has a gas cost of 200. Access to a local variable is just a `POP`, with a cost of 2 gas.